src/modules/content-seeding/news-aggregator.js

```diff
@@ -0,0 +1,290 @@
+			return insertIntoFeeds(seedUrl);
+		}
+		return [ { rowCount: 0 } ];
+	})
+	.then(async (response) => {
+		// response: [{.., rowCount: _, ...}, ..]
+		if (response[0].rowCount === 1) {
+			winston.info('Insert operation performed successfully.');
+		}
+		await truncateArticlesTable();
+		// feeds: [{url: _, mtbu: _, lastrequesttime: _, lastupdatetime: _}, .....]
+		const feeds = await filterFeeds();
+		if (feeds.length > 0) {
+			const parseFeedPromises = [];
+			for (const feed of feeds) {
+				/*
+					CHECK FOR NEW ARTICLES:
+						- checks:
+							-> shouldn't be older than 1 week.
+							-> Avoid duplicate stroies from the same feed.
+							-> To avoid duplicate stories, compare article's pubdate with the lastupdatetime for that feed.
+				*/
+				const newArticles = [];
+				let latestArticleDate = feed.lastupdatetime;
+				parseFeedPromises.push(
+					parseNewsFeed(feed.url)
+					.then((newsArticles) => {
+						const promises = [];
+						for (const newsArticle of newsArticles) {
+							const articlePubDate = (new Date(newsArticle.published)).getTime();
+							const sevenDays = 7 * 24 * 60 * 60 * 1000;
+							if (hasNewNewsArticle(articlePubDate, feed.lastupdatetime) && articleAge(articlePubDate) <= sevenDays) {
+								latestArticleDate = Math.max(articlePubDate, latestArticleDate);
+								const article = _.pick(newsArticle, [ 'title', 'link', 'content', 'published' ]);
+								newArticles.push(article);
+							}
+						}
+
+						if (newArticles.length > 0) {
+							const mtbu = calculateMTBU(newArticles);
+							// update MTBU, lastrequesttime, lastupdatetime
+							promises.push(
+								updateFeedsOnNewArticles(latestArticleDate, feed.url, mtbu)
+							);
+
+							// Insert new articles
+							for (const article of newArticles) {
+								const url = article.link;
+								const articleObj = article;
+								const contentForLexemes = article.title + ' ' + article.content;
+								promises.push(
+									insertNewArticles(url, articleObj, contentForLexemes)
+								);
+							}
+						} else {
+							// update lastrequesttime
+							promises.push(
+								updateFeedsOnNoNewArtcles(feed.url)
+							);
+						}
+
+						return Promise.all(promises);
+					})
+				);
+			}
+			await Promise.all(parseFeedPromises);
+			const roomSpecificNewsArticles = await getRoomSpecificNews();
+			return roomSpecificNewsArticles;
+		}
+		return feeds;
+	});
+};
+
+
+// Invoke this job every hour
+setInterval(async () => {
+	try {
+		const roomSpecificNewsArticles = await newsAggregator();
+		if (roomSpecificNewsArticles.length > 0) {
+			const latestNewsForRooms = getMostRecentNewsForRoom(roomSpecificNewsArticles);
+			const threads = buildThreads(latestNewsForRooms);
+			bus.emit('change', threads);
+		} else {
+			winston.info('No new news stories found !!');
+		}
+	} catch (e) {
+		winston.error(`somethong went wrong while aggregating news, ${e}`);
+	}
+}, 3600000);
```

Comment:
Also, splitting it up as e.g. 60 * 60 * 1000 for one hour makes it easier to read. There is no performance impact, JS engines are smart enough to pre-compute constant expressions.